pkg/inspector/image-inspector.go

```diff
@@ -210,27 +223,33 @@ func (i *defaultImageInspector) pullImage(client *docker.Client) error {
 		return authCfgErr
 	}
 
-	reader, writer := io.Pipe()
-	// handle closing the reader/writer in the method that creates them
-	defer writer.Close()
-	defer reader.Close()
-	imagePullOption := docker.PullImageOptions{
-		Repository:    i.opts.Image,
-		OutputStream:  writer,
-		RawJSONStream: true,
-	}
-
-	bytesChan := make(chan int)
-	go aggregateBytesAndReport(bytesChan)
-	go decodeDockerPullMessages(bytesChan, reader)
-
 	// Try all the possible auth's from the config file
 	var authErr error
 	for name, auth := range imagePullAuths.Configs {
-		if authErr = client.PullImage(imagePullOption, auth); authErr == nil {
+		reader, writer := io.Pipe()
+		// handle closing the reader/writer in the method that creates them
+		defer writer.Close()
+		defer reader.Close()
+		imagePullOption := docker.PullImageOptions{
+			Repository:    i.opts.Image,
+			OutputStream:  writer,
+			RawJSONStream: true,
+		}
+
+		bytesChan := make(chan int, 100)
+		parsedErrors := make(chan bool, 1)
+		go decodeDockerPullMessages(parsedErrors, bytesChan, reader)
+
+		if authErr := client.PullImage(imagePullOption, auth); authErr != nil {
+			return authErr
+		}
+		writer.Close()
```

Comment:
strange here that you're deferring reader and writer close above but explicitly closing them here.  Why?